src/main/daml/Daml/Finance/Lifecycle/Rule/Util.daml

```diff
@@ -0,0 +1,48 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Lifecycle.Rule.Util where
+
+import DA.Either (partitionEithers)
+import DA.List (head, groupOn)
+import DA.Time
+import DA.Date
+import Daml.Finance.Interface.Claims.Types (Pending(..))
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as Instrument (Q, qty)
+
+-- | Merge consumed and produced instruments into a list of pending settlements.
+-- This will only reproduce instrument and quantity, not tag or time.
+mergeConsumedAndProduced : [Instrument.Q] -> [Instrument.Q] -> [Pending]
```

Comment:
I would suggest to define a separate `Pending` type to be used in this context rather than the one used in Claims.
This has a few advantages:
- no dependency on `Interface.Claims`
- no meaningless `tag` and `t` attributes (which are well-defined only in the context of claims)

Reply:
Good point, thanks!